src/db.rs

```diff
@@ -1,25 +1,19 @@
-use diesel::prelude::*;
-use diesel::pg::PgConnection;
 use std::env;
-use diesel::r2d2::{ Pool, PooledConnection, ConnectionManager };
-use r2d2_postgres::PostgresConnectionManager;
+use diesel::r2d2::ConnectionManager;
+use r2d2::Pool;
+use diesel::pg::PgConnection;
 
-pub type PgPool = Pool<ConnectionManager<PgConnection>>;
-pub type PgPooledConnection = PooledConnection<ConnectionManager<PgConnection>>;
+pub type PgPool = r2d2::Pool<ConnectionManager<PgConnection>>;
 
 fn init_pool(database_url: &str) -> PgPool {
-    let manager = PostgresConnectionManager::new(database_url);
-    let pool = r2d2::Pool::new(manager).unwrap();
-    pool
+    let manager = ConnectionManager::<PgConnection>::new(database_url);
```

Comment:
Which version of `ConnectionManager` you were using here was one of the big problems, I think?